### PR TITLE
test: update test262 parser test

### PR DESCRIPTION
**Pull request by fisker**
Just copied from [acorn](https://github.com/acornjs/acorn/blob/12ad1164a70dbe804f1ece822f34d100545f1afa/bin/run_test262.js)

### CODE REVIEW

**Comment by fisker** [comment]
It takes me a while, but we do have lots of tests can't pass in class fields/methods.

File: test/test262-parser-tests/unsupported-features.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+explicit-resource-management
+regexp-modifiers
+import-assertions
+import-attributes
+regexp-duplicate-named-groups
+regexp-v-flag
+arbitrary-module-namespace-names
+source-phase-imports
+source-phase-imports-module-source
```
Comment (fisker) [inline test/test262-parser-tests/unsupported-features.txt:9]: @3cp Can you confirm this is the correct list of features that we don't support?

Comment (3cp) [reply test/test262-parser-tests/unsupported-features.txt:9]: Regexp modifiers should work.
Meriyah doesn't parse regexp internal, the whole regex is one tokenValue, parsed by native RegExp provided by the runtime (which runs meriyah), `try { return new RegExp(...) } catch ...`.

In same sense, regexp-duplicate-named-groups properly also works.

**Comment by 3cp** [comment]
Don't know what's wrong with mocha in nodejs v22 on CI, I don't have the issue locally.

**Comment by 3cp** [comment]
The whitelist is ones we have trouble with?

File: test/test262-parser-tests/unsupported-features.txt (hunk 1 above)
Comment (3cp) [reply test/test262-parser-tests/unsupported-features.txt:9]: class-static-block is not supported.

Comment (3cp) [reply test/test262-parser-tests/unsupported-features.txt:9]: class-static-methods-private is not supported.

**Comment by fisker** [comment]
> Don't know what's wrong with mocha in nodejs v22 on CI, I don't have the issue locally.

https://github.com/nodejs/node/issues/53902



> The whitelist is ones we have trouble with?

Yes.

File: test/test262-parser-tests/unsupported-features.txt (hunk 1 above)
Comment (fisker) [reply test/test262-parser-tests/unsupported-features.txt:9]: > Meriyah doesn't parse regexp internal

I didn't know that at first, I think it's really bad to rely on runtime behavior, maybe we should do something in future.

**Review by 3cp — APPROVED** [review]

**Pull request merged**